src/senaite/jsonapi/api.py

```diff
@@ -1086,6 +1099,32 @@ def is_creation_allowed(portal_type, container):
     return True
 
 
+def is_update_allowed(obj):
+    """Returns whether the update of the object passed in is supported
+
+    :param obj: The object to be updated
+    :type obj: ATContentType/DexterityContentType
+    :returns: True if it is allowed to update this object
+    :rtype: bool
+    """
+    # Do not allow the update of objects that belong to site root folder
+    parent = api.get_parent(obj)
+    if api.is_portal(parent):
+        return False
+
+    # Do not allow the update of objects that belong to setup folder
+    if parent == api.get_setup():
+        return False
+
+    # Look for an update-specific adapter for this object
+    adapter = queryAdapter(obj, IUpdate)
+    if adapter:
+        return adapter.is_update_allowed()
+
+    # We do not support the update of inactive objects
+    return api.is_active(obj)
```

Comment:
Why not? It is also possible in the UI.

Reply:
> Why not? It is also possible in the UI.

Ok, agree. This should be controled by core, as part of the workflow permissions mapping rather than explicitly here